### PR TITLE
Created merchant show action, edited route, merchant show tests are passing

### PR DESCRIPTION
**Pull request by emaust**

### CODE REVIEW

**Review by tofuandeve — APPROVED** [review]
Nice work!

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   root :to => "homepages#index"
+  
+  
+  resources :merchants, except: [:index, :edit, :update]
```
Comment (michaela260) [inline config/routes.rb:6]: yay :)

File: app/controllers/merchants_controller.rb [hunk 2]
```diff
@@ -1,2 +1,16 @@
 class MerchantsController < ApplicationController
+
+    def show 
+        @merchant = Merchant.find_by(id: params[:id])
+
+        if @merchant.nil?
+            flash[:error] = "You must log in to view the merchant dashboard."
```
Comment (michaela260) [inline app/controllers/merchants_controller.rb:7]: we should choose names for the flash message categories

File: test/controllers/merchants_controller_test.rb [hunk 3]
```diff
@@ -1,7 +1,27 @@
...
-  # it "does a thing" do
-  #   value(1+1).must_equal 2
-  # end
-end
+
+  describe "show" do
+
+        it "correctly locates a merchant by username" do
+          valid_merchant = Merchant.create(username: "industrious_raccoon", email: "givemeyoureggshells@bandit.com")
+          valid_merchant = Merchant.find_by(username: valid_merchant.username)
```
Comment (michaela260) [inline test/controllers/merchants_controller_test.rb]:
```suggestion
```

File: test/controllers/merchants_controller_test.rb [hunk 4]
```diff
@@ -1,7 +1,27 @@
...
 
 describe MerchantsController do
-  # it "does a thing" do
-  #   value(1+1).must_equal 2
-  # end
-end
+
+  describe "show" do
+
+        it "correctly locates a merchant by username" do
```
Comment (michaela260) [inline test/controllers/merchants_controller_test.rb]:
```suggestion
        it "successfully redirects to the show page for a valid merchant" do
```

File: test/controllers/merchants_controller_test.rb [hunk 5]
```diff
@@ -1,7 +1,27 @@
...
+          valid_merchant = Merchant.find_by(username: valid_merchant.username)
+
+          get merchant_path(valid_merchant.id)
+          must_redirect_to merchant_path(valid_merchant.id)
+        end
+
+        it "redirects to root path when given an invalid id" do
+          
+          get merchant_path(-1)
+          must_redirect_to root_path
```
Comment (michaela260) [inline test/controllers/merchants_controller_test.rb:17]: we should make sure this and current method go to the same route

**Review by michaela260 — APPROVED** [review]
nice! see small changes :)

**Pull request merged**